controllers/pod.go

```diff
@@ -71,24 +87,39 @@ func (r *SingleClusterReconciler) getRollingRestartTypeMap(
 		}
 
 		podStatus := r.aeroCluster.Status.Pods[pods[idx].Name]
-		if addedNSDevices == nil && podStatus.AerospikeConfigHash != requiredConfHash {
-			// Fetching all block devices that have been added in namespaces.
-			addedNSDevices, err = r.getNSAddedDevices(rackState)
-			if err != nil {
-				return nil, err
+		if podStatus.AerospikeConfigHash != requiredConfHash {
+			if addedNSDevices == nil {
+				// Fetching all block devices that have been added in namespaces.
+				addedNSDevices, err = r.getNSAddedDevices(rackState)
+				if err != nil {
+					return nil, nil, err
+				}
+			}
+
+			// If dynamic commands have failed in previous retry, then we should not try to update config dynamically.
+			if !podStatus.DynamicConfigFailed {
+				// Fetching all dynamic config change.
+				dynamicConfDiffPerPod[pods[idx].Name], err = r.handleDynamicConfigChange(rackState, pods[idx])
+				if err != nil {
+					return nil, nil, err
+				}
 			}
 		}
 
-		restartType := r.getRollingRestartTypePod(rackState, pods[idx], confMap, addedNSDevices)
+		restartTypeMap[pods[idx].Name] = r.getRollingRestartTypePod(rackState, pods[idx], confMap, addedNSDevices,
+			len(dynamicConfDiffPerPod[pods[idx].Name]) > 0)
 
-		restartTypeMap[pods[idx].Name] = restartType
+		if podStatus.DynamicConfigFailed {
+			restartTypeMap[pods[idx].Name] = mergeRestartType(restartTypeMap[pods[idx].Name], quickRestart)
+		}
```

Comment:
No, if hash is same, then may be user has reverted the changes after dynamic failure happened. In that case if DynamicConfigFailed is true then instead of no-restart, quick restart should happen.